frontends/common/options.cpp

```diff
@@ -43,7 +58,9 @@ CompilerOptions::CompilerOptions() : Util::Options(defaultMessage) {
                        exit(0); return false; }, "Print compiler version");
     registerOption("-I", "path",
                    [this](const char* arg) {
-                       preprocessor_options += std::string(" -I") + arg; return true; },
+                       char buffer[PATH_MAX];
+                       convertToAbsPath(arg, buffer);
```

Comment:
this does not seem to change the files, but rather the include paths. I'm not sure this solves the issues you are seeing. If you are targeting the p4 14 to 16 converter (https://github.com/p4lang/p4c/blob/master/frontends/p4/fromv1.0/programStructure.cpp#L469) hardcodes the path already to a compile-time defined path.